### PR TITLE
Add support for allowEmptyValue

### PR DESCRIPTION
**Pull request by Adrien-Martins**
See http://swagger.io/specification/#parameterObject

### CODE REVIEW

File: lib/validator.js [hunk 1]
```diff
@@ -99,6 +99,10 @@ module.exports = function validator(options) {
                 schema = schema.required();
             }
 
+            if (parameter.in !== 'body' && parameter.allowEmptyValue){
+              schema = schema.allow('');
```
Comment (tlivings) [inline lib/validator.js]: Wouldn't this be `schema.optional()`?

Comment (Adrien-Martins) [inline lib/validator.js]: schema.optional() seems to allow *undefined* values but they don't allow empty string value.
I use the solution given here: https://github.com/hapijs/joi/issues/482

Maybe I can also add optional(), but I don't think that it is possible to send an undefined value in an HTTP request. Do you want that I add it too ?

Comment (tlivings) [inline lib/validator.js]: Ok sounds fine then.

**Comment by tlivings** [comment]
:+1:

**Comment by subeeshcbabu** [comment]
LGTM

**Pull request merged**